src/bp-core/bp-core-catchuri.php

```diff
@@ -1341,7 +1341,13 @@ function bp_private_network_template_redirect() {
 					$actual_link = ( is_ssl() ? 'https://' : 'http://' ) . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
 
 					if ( $actual_link !== wc_lostpassword_url() && ! bp_is_activation_page() ) {
-						if ( 'yes' !== get_option( 'woocommerce_enable_myaccount_registration' ) && $id !== intval( get_option( 'woocommerce_myaccount_page_id' ) ) ) {
+						if (
```

Comment:
@milanhirapra Solution is correct. But I think you should add a filter here and 3rd party plugin support should add using a filter so in the future if it conflicts with another plugin we can add support using the filter.

Reply:
@dipskakadiya I added a new file for Woocommerce compatibility. Please check and review the code.